proposals/transcoding_invalid_identifiers/README.md

```diff
@@ -0,0 +1,321 @@
+- **Simplicity**: The encoding and decoding algorithms are reasonably simple to implement.
+
+# Proposed solution
+
+Defined in [RFC-3492](https://datatracker.ietf.org/doc/html/rfc3492), we find PunyCode. PunyCode is a specialization
+of Bootstring algorithms. Bootstring would perform better than base encodings and url encodings. 
+
+However, a basic Punycode implementation have an initial limitations for OpenUSD.
+- Punycode (and Bootstring in general), involves an initial basic code segregation. In the definition of Punycode, 
+basic codes are all ASCII characters whose value is less than or equal to the parameter `initial_n`. For example, 
+`--> $1.00 <--` will be converted to `--> $1.00 <--` in Punycode (i.e., no change), however this identifier is 
+invalid. An implementation of Punycode (or another Bootstring algorithm) would require a specific `IsBasicCode` function to 
+account for this, also adapting different steps in the algorithm to account for the situation where `initial_n` is 
+non existent (i.e. 0).
+- Another less impactful problem with Punycode (but not with Bootstring), is the way the extended code characters 
+are encoded/decoded. To encode, Punycode uses base 36 encoding by default. This is due to the fact that upper case and
+lower case are considered the same. We could lift this limitation and use less memory when encoding. 
+- Finally, just as other encodings, special care must be taken to address leading digits. In the case of Punycode, 
+we can treat any leading digits as extended codes and add them after the delimiter.
+
+These concerns can be addressed with a custom implementation of Bootstring. This strategy has the following advantages:
+- **Efficiency**, 100% for basic code characters, at worst 72% for extended code characters. This is because variable 
+length encoding is way more efficient than simple bit shifting encoding (i.e. base 62).
+- **Readability**, a valid identifier will be encoded without any change, i.e. `hello` will be encoded to `hello`; 
+non-valid identifiers consisting mostly of valid characters will be partially encoded, i.e. `hello world` will be 
+encoded to `tn__helloworld_lA`, i.e. only the space is encoded; and only non-valid identifiers consisting mostly 
+of invalid characters will be non-readable, i.e. `->$.<-` will be encoded to `tn__a0I26g1D`. 
+This is improved over other encoding methods, i.e. base62, where every case is obfuscated.
+- **Querying**, for basic code characters, querying is the same as before.
+
+Disadvantages:
+- **Querying**, unfortunately encoding the search term and doing character comparison will not work as this is not a 
+byte-aligned encoding.This will require all paths to be decoded as they are traversed.
+- **Prefix**, to give a hint of transcoding we add a prefix `tn__`, in similar fashion to `xn--` in Punycode. 
+For short identifiers this may represent a big overhead.
```

Comment:
suggestion-- "or potentially collide with identifiers starting with `tn__` for reasons other than a hint for decoding".

Reply:
Added